src/main.rs

```
@@ -135,32 +138,37 @@ async fn fetch_data() -> Result<String, String> {
 
 #[cfg(test)]
 mod tests {
+    use pretty_assertions::assert_eq;
     use reqwest::StatusCode;
 
     use crate::requests::DataRequest;
 
     async fn post_data_request() -> DataRequest {
         let json = include_bytes!("../docs/examples/data_request.json");
-        let response = reqwest::Client::new()
+        let response_result = reqwest::Client::new()
             .post(format!("http://localhost:8080/requests"))
             .json(&serde_json::from_slice::<serde_json::Value>(json).unwrap())
             .send()
-            .await
-            .unwrap();
-        assert_eq!(response.status(), StatusCode::CREATED);
+            .await;
+        assert_eq!(response_result.is_ok(), true, "POST response (from /requests) was not OK");
```

I dont really get how this improves the code. Unwraping/panicing in tests is totally fine. This assert will just discard the actual error message. If you want to add a more descriptive error message you could use `Result::expect`.

---

Done.